sw/device/lib/dif/register.h

```diff
@@ -0,0 +1,129 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_DIF_REGISTER_H_
+#define SW_DEVICE_LIB_DIF_REGISTER_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/**
+ * A reg32_t is an opaque handle to an MMIO register; it should only be modified
+ * using the functions provided in this header.
+ */
+typedef struct reg32 {
+  volatile uint32_t *inner_ptr;
+} reg32_t;
+
+/**
+ * Create a new |reg32_t| from the given address.
+ *
+ * @param address an address to an MMIO register.
+ * @return a |reg32_t| value representing that register.
+ */
+inline reg32_t reg32_new(uintptr_t address) {
```

Comment:
Could you please in the commit message, or maybe at the top of this header file expand on the implementation decisions, rationale and why it is better than alternatives. The approach taken in register.c,.h

Reply:
Done.